### PR TITLE
Add utility scripts for simplifying changes to image.yaml files and context directories

### PR DESCRIPTION
**Pull request by tmckayus**
This script makes the necessary changes to the image.*.yaml files
to download a new version of the oshinko CLI. It regenerates the
docker build directories and commits the changes to a release
branch created for the new version. This is to make the release
process easier.

### CODE REVIEW

File: release.sh [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    echo "  -h                  Show this message"
+}
+
+if [ "$#" -lt 1 ]; then
+    usage
+    exit 1
+fi
+
+# Set a default for the hadoop version
+HVER=2.7
```
Comment (crobby) [inline release.sh]: With no way to override (and we probably don't really need one) this seems more than just a default

File: release.sh [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    echo "Changes the image.*.yaml files and regenerates the build directories"
+    echo 'Commits the changes on a branch release_$RELEASE, ready for a push'
+    echo
+    echo "Usage: release.sh [options]"
+    echo
+    echo "required arguments"
+    echo
+    echo "  -r RELEASE          The oshinko release number, like v0.4.4"
+    echo
+    echo "optional arguments:"
```
Comment (crobby) [inline release.sh]: We specify the default for OSHINKO_VERSION, but the others do not list their default values.  Should probably include a note about what happens if they are not specified.

File: README.md [hunk 3]
```diff
@@ -17,25 +17,53 @@ The easiest way to build the s2i images is to use the makefiles provided:
     $ make -f Makefile.java
     $ make -f Makefile.scala
 
-The default repository for the image can be set with the `LOCAL_IMAGE` var:
+The default repository for the image can be overridden with the `LOCAL_IMAGE` var:
 
     $ LOCAL_IMAGE=myimage make -f Makefile.pyspark
 
-## Remaking Docker context directories when things change
+## Modifying dependencies in the image.*.yaml files
```
Comment (tmckayus) [inline README.md]: bah, * here is turning on bolding. I'll fix that

File: change-yaml.sh [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    echo
+    echo "  -l SCALA_VERSION    The scala version, like 2.11.8. Applies to the scala image"
+    echo
+    echo "  -t SBT_VERSION      The sbt version, like 0.13.13. Applies to the scala image"
+    echo
+    echo "  -h                  Show this message"
+}
+
+if [ "$#" -eq 0 ]; then
+    echo No options specified, changing nothing.
```
Comment (tmckayus) [inline change-yaml.sh]: probably should add an exit here

File: change-yaml.sh [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+	# Fix the url references
+	sed -i "s@https://archive.apache.org/dist/spark/spark-.*/spark-.*-bin-@https://archive.apache.org/dist/spark/spark-${SPARK}/spark-${SPARK}-bin-@" image.*.yaml
+
+	# Fix the md5 sum references on the line following the url
+        sed -i '\@url: https://archive.apache.org/dist/spark/@!b;n;s/md5.*/md5: '$sum'/' image.*.yaml
+
+        # For the scala image, the openshift-spark base image is based on the first 2 characters in the spark version
+        BV=$(echo ${SPARK} | cut -d'.' -f1,2)
+        sed -i "s@radanalyticsio/openshift-spark:.*-latest@radanalyticsio/openshift-spark:${BV}-latest@" image.*.yaml
```
Comment (tmckayus) [inline change-yaml.sh]: this could just run on image.scala.yaml instead of all 3

**Pull request merged**